lib/util/stream_utils.js

```diff
@@ -523,51 +523,69 @@ shaka.util.StreamUtils = class {
    * and assigns it to the given variant.
    * If that query has been done before, instead return a cached result.
    * @param {!shaka.extern.Variant} variant
-   * @param {!MediaDecodingConfiguration} decodingConfig
+   * @param {!Array.<!MediaDecodingConfiguration>} decodingConfigs
    * @private
    */
-  static async getDecodingInfosForVariant_(variant, decodingConfig) {
-    const cacheKey =
-        shaka.util.StreamUtils.alphabeticalKeyOrderStringify_(decodingConfig);
-
-    const cache = shaka.util.StreamUtils.decodingConfigCache_;
-    if (cache[cacheKey]) {
-      shaka.log.v2('Using cached results of mediaCapabilities.decodingInfo',
-          'for key', cacheKey);
-      variant.decodingInfos.push(cache[cacheKey]);
-    } else {
-      // Do a final pass-over of the decoding config: if a given stream has
-      // multiple codecs, that suggests that it switches between those codecs
-      // at points of the go-through.
-      // mediaCapabilities by itself will report "not supported" when you
-      // put in multiple different codecs, so each has to be checked
-      // individually. So check each and take the worst result, to determine
-      // overall variant compatibility.
-      const results = await shaka.util.StreamUtils
-          .checkEachDecodingConfigCombination_(decodingConfig);
-      if (results && results.length) {
-        /** @type {?MediaCapabilitiesDecodingInfo} */
-        let overallResult = null;
-        for (const result of results) {
-          if (!overallResult) {
-            overallResult = result;
-          } else {
-            overallResult.supported =
-                overallResult.supported && result.supported;
-            overallResult.powerEfficient =
-                overallResult.powerEfficient && result.powerEfficient;
-            overallResult.smooth = overallResult.smooth && result.smooth;
-            if (!result.keySystemAccess) {
-              overallResult.keySystemAccess = null;
-            }
-          }
-        }
-        if (overallResult) {
-          cache[cacheKey] = overallResult;
-          variant.decodingInfos.push(overallResult);
+  static async getDecodingInfosForVariant_(variant, decodingConfigs) {
+    /**
+     * @param {?MediaCapabilitiesDecodingInfo} a
+     * @param {!MediaCapabilitiesDecodingInfo} b
+     * @return {!MediaCapabilitiesDecodingInfo}
+     */
+    const merge = (a, b) => {
+      if (!a) {
+        return b;
+      } else {
+        const res = shaka.util.ObjectUtils.cloneObject(a);
+        res.supported = a.supported && b.supported;
+        res.powerEfficient = a.powerEfficient && b.powerEfficient;
+        res.smooth = a.smooth && b.smooth;
+        if (!b.keySystemAccess) {
+          res.keySystemAccess = null;
```

Comment:
We're getting the `REQUESTED_KEY_SYSTEM_CONFIG_UNAVAILABLE` error because `decodingInfo.keySystemAccess` is `null` for every variant. I think this block is the reason. `res.keySystemAccess` can be set to null here, but never a valid value. If I make this change, it gets us past the `REQUESTED_KEY_SYSTEM_CONFIG_UNAVAILABLE` error.

```suggestion
        res.keySystemAccess = b.keySystemAccess ? b.keySystemAccess : null;
```